src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -224,6 +224,19 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                     {
                         case "-": // csi -- script.csx
                             if (value != null) break;
+                            if (arg == "-")
+                            {
+                                if (Console.IsInputRedirected)
+                                {
+                                    sourceFiles.Add(ToCommandLineSourceFile("stdin" + ScriptFileExtension, isInputRedirected: true));
```

Comment:
>"stdin" + ScriptFileExtension [](start = 76, length = 29)

Is it necessary to give it a different name and add an extension? Or could we just use `new CommandLineSourceFile("-", isScriptFile: true, isInputRedirected: true)` instead?

Reply:
It is not necessary and I do not feel strongly about this choice. I just felt it would be more readable in Error List / logs. Also, if we do not provide the output file name, it creates an executable called `stdin.exe` (or `stdin` in case of Unix).

However, on a downside, chances of name collisions are higher for former in `stdin.cs` vs. `-`. In fact, I was trying to create an AppDomain (for net472) in VB's test, and it gave me error that there already is an assembly loaded with the name `stdin.dll` (wasn't the case in an identical C# test), therefore i gave output name in all test cases (but later removed AppDomain altogether as it wasn't supported in netcore).

I will change them to `-`